include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,752 @@
+ *
+ * Please note it is always recommended that application using async mode,
+ * provide sufficiently large buffer size to avoid ODP_COMP_ERR_OUT_OF_SPACE.
+ * Else it is recommended that application set required user context on packet
+ * before calling each odp_comp_op_enq() to correctly retrieve required
+ * information associated with packet.
+ *
+ * @param         pkt_in   Input Packet
+ * @param         pkt_out  Output Packet
+ * @param         param    Operation parameter
+ *
+ * @return 0 on success
+ * @retval <0 on failure
+ */
+int odp_comp_op_enq(const odp_packet_t pkt_in,
+		    const odp_packet_t pkt_out,
+		    const odp_comp_packet_op_param_t *param);
+
+/**
+ * Comp API to process multiple packets synchronously
+ *
+ * If session is created in ODP_COMP_SYNC mode, this call wait for operation
+ * to complete.
+ *
+ * If session is created in ODP_COMP_ASYNC mode, behavior is undefined.
+ *
+ * for compression + hash, call returns with hash appended at the end of
+ * packet on which odp_packet_op_param_t:last field is set.
+ *
+ * User should compute processed data len = output_data_range.len - digest_len,
+ * where digest_len queried through odp_comp_hash_alg_capability().
+ *
+ * For every input packet entry in in_pkt array, user should pass
+ * corresponding valid output packet handle. If during processing of packet,
+ * corresponding buffer goes out of space, then API returns with number of
+ * entries successfully processed.
+ *
+ * A successful proceesed entry means input packet has been completely consumed
+ * and whole output has been written into output packet and corresponding
+ * result entry is updated with offset:length information and error code set to
+ * ODP_COMP_ERR_NONE.
+ *
+ * In case of partially processed array i.e. when
+ * number of packets returned < num_pkt, application should check on result
+ * on 1st failed entry (i.e. n+1th entry where
+ * n = number of entried returned) to ensure if error code is
+ * ODP_COMP_ERR_OUT_OF_SPACE.
+ *
+ * If status code on 1st failed entry set to ODP_COMP_ERR_OUT_OF_SPACE,
+ * application should call API again with corresponding output packet
+ * available with free space.
+ *
+ * All arrays should be of num_pkt size.
+ *
+ * @param         pkt_in   Input Packets
+ * @param         pkt_out  Output Packet to store output of corresponding entry
+ * @param         param    Operation parameters array
+ * @param[out]    result   Result array corresponding to each entry
+ * @param         num_pkt  Number of packets to be processed
+ *
+ * @return Number of input packets consumed (0 ... num_pkt)
+ * @retval <0 on failure
+ */
+int odp_comp_op_multi(const odp_packet_t pkt_in[],
+		      const odp_packet_t pkt_out[],
+		      odp_comp_packet_op_param_t param[],
+		      odp_comp_packet_op_result_t result[],
+		      int num_pkt);
+
+/**
+ * Comp API to process multiple packets asynchronously
+ *
+ * Performs the ASYNC decompression operations on the packet array.
```

Comment:
I guess this function can do both compression and decompression asynchronously. This line can be removed.